src/core/db/src/persistent_data_interface.cpp

```diff
@@ -8,18 +8,85 @@ sqlite3pp::database *database::DeviceDbInterface::db = nullptr;
 
 bool database::DeviceDbInterface::connect(std::string db_file_path) {
   db = new sqlite3pp::database(db_file_path.c_str());
+  db->execute("CREATE TABLE devices ("
+              "id TEXT PRIMARY KEY,"
+              "link TEXT NOT NULL UNIQUE, type TEXT NOT NULL,"
+              "title TEXT NOT NULL, description TEXT NOT NULL) ");
 
   std::cout << "db file created" << std::endl;
 }
 
 bool database::DeviceDbInterface::disconnect() { delete db; }
 
-bool database::DeviceDbInterface::addDevice(std::string mac, std::string ip,
-                                            std::string port) {
-  sqlite3pp::command cmd(
-      *db, "INSERT INTO devices (mac, ip, port) VALUES (?, ?, ?)");
-  cmd.binder() << mac << ip << port;
+bool database::DeviceDbInterface::add(std::string id, std::string link,
+                                      std::string type, std::string title,
+                                      std::string description) {
+
+  sqlite3pp::command cmd(*db, "INSERT INTO devices (id, link, type, title, "
+                              "description) VALUES (?, ?, ?, ?, ?)");
+  cmd.binder() << id << link << type << title << description;
+  cmd.execute();
+  return true;
+}
+
+std::vector<radlet_dock::device::Device> database::DeviceDbInterface::getAll() {
```

Comment:
Now consider how many times vector constructor will be called,
1st : line 33 (no issues since an empty vector)
2nd: line 56 (time equal to number of elements in vector)
3rd: In caller method while copying result (costly as line 56)

Solution would be to return shared_ptr or use forwarding reference.